src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvThread.cs

```diff
@@ -103,15 +102,15 @@ public Task StartAsync()
             {
                 var stepTimeout = TimeSpan.FromTicks(timeout.Ticks / 3);
 
-                Post(t => t.AllowStop());
+                await PostAsync(t => t.AllowStop(), this).ConfigureAwait(false);
```

Comment:
What's the benefit of PostAsync here if we are waiting for _threadTcs.Task to complete anyway?